### ISSUE TITLE
hs.GetActionsByInterface(plugin.Id) returns empty list

### ISSUE THREAD

**Issue by alexbk66** [HomeSeer/Plugin-SDK#145]
_hs.GetActionsByInterface(plugin.Id)_ returns empty list
_hs.GetAllEvents()_ does something weird - doesn't return, but try/catch don't catch any exception
_hs.GetEventByRef()_ too

> /// PLEASE NOTE: This was ported directly from the legacy HS3 API and has not been fully reviewed to ensure
> ///  proper compatibility and support through this SDK.  This may undergo significant change in the near future.
> ///  Please use with caution.

This comment probably means it wasn't tested at all?

[EDIT]

I also tried the old HS3 way:

```
IPlugInAPI.strTrigActInfo[] acts1 = hs3Callback.GetActions(plugin.Id);
IPlugInAPI.strTrigActInfo[] acts2 = hs3Callback.GetActions(plugin.Name);
```

The first call returns empty list, the second one works.  So I changed to

```
hs.GetActionsByInterface(plugin.Name) 
```
And it works

**Comment by spudwebb** [comment]
For GetActionsByInterface(), you're right you need to pass the plugin name, this is the same problem as #141

I will have a look at GetAllEvents() and GetEventByRef()

**Comment by spudwebb** [comment]
GetAllEvents and GetEventByRef do not work because  the EventData struct is not marked as Serializable. This will be fixed in next plugin SDK. (PSDK-122)

**Comment by alexbk66** [comment]
> For GetActionsByInterface(), you're right you need to pass the plugin name

I realised that - hard way - wasting a day of my life!

I think this should be changed to plugin ID?  And the function argument and comments says pluginID:
```
/// <param name="pluginId">The ID of the plugin</param>
/// <returns>A list of all of the actions managed by the plugin</returns>
List<TrigActInfo> GetActionsByInterface(string pluginId);

```

**Comment by alexbk66** [comment]
Also

```
 /// <param name="**pluginName**">The **ID of the plugin** that owns the trigger type</param>
 /// <param name="trigId">The ID of the trigger type</param>
 /// <returns>An array of trigger data</returns>
 TrigActInfo[] GetTriggersByType(string pluginName, int trigId);
```

Mess

**Comment by mnsandler** [comment]
there is definitely a problem. see me other post 
https://github.com/HomeSeer/Plugin-SDK/issues/148#issuecomment-769818929

**Comment by spudwebb** [comment]
The following methods have been fixed in Plugin SDK version 1.2.2
```
EventGroupData GetEventGroupById(int groupRef);
List<EventGroupData> GetAllEventGroups();
List<EventData> GetAllEvents();
EventData GetEventByRef(int eventRef);
List<EventData> GetEventsByGroup(int groupId);
```

For the other problems related to the mix up between plugin ID and plugin name, please use issues #141 and #148
